Source/Evolve/WarpXEvolve.cpp

```diff
@@ -241,6 +241,7 @@ WarpX::Evolve (int numsteps)
             mypc->SortParticlesByBin(sort_bin_size);
         }
 
+        if (warpx_py_beforeEsolve) warpx_py_beforeEsolve();
```

Comment:
This needs to be inside the if block, because the Electromagnetic case has already done the Esolve at this point (it's in `OneStep_nosub`, so we only want this for electrostatic.